app/main.py

```diff
@@ -1,66 +1,64 @@
-def add_book(book: schemas.Book, session: Session = Depends(get_session)):
-    response = controller.add_book(book, session)
-    if type(response) == Err:
-        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR.value)
+@api_router.post(
+    "/books/",
+    tags=["books"],
+    response_model=models.Book,
+    status_code=status.HTTP_201_CREATED,
+)
+async def add_book(book: models.BookIn):
+    response = await controller.add_book(SQLiteRepository, book)
     return response
 
 
-@api_router.get("/books/{book_id}", tags=["books"])
-def get_book(book_id: int, session: Session = Depends(get_session)):
-    response = controller.fetch_a_book(book_id, session)
-    if type(response) == Err and response.error == NO_OBJECT_ERROR:
-        raise HTTPException(status_code=HTTPStatus.NOT_FOUND.value)
-    elif type(response) == Err:
-        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR.value)
+@api_router.get("/books/{book_id}", tags=["books"], response_model=models.Book)
+async def get_book(book_id: int):
+    response = await controller.fetch_a_book(SQLiteRepository, book_id)
+    if isinstance(response, NotFoundError):
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND.value, detail="Book with such ID not found"
+        )
     return response
 
 
-@api_router.put("/books/{book_id}", tags=["books"])
-def update_book(
-    book_id: int, book: schemas.Book, session: Session = Depends(get_session)
-):
-    response = controller.update_book(book_id, book, session)
-    if type(response) == Err and response.error == NO_OBJECT_ERROR:
-        raise HTTPException(status_code=HTTPStatus.NOT_FOUND.value)
-    elif type(response) == Err:
-        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR.value)
+@api_router.delete("/books/{book_id}", tags=["books"])
+async def delete_book(book_id: int):
+    response = await controller.delete_book(SQLiteRepository, book_id)
+    if response == 0:
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND.value, detail="Book with such ID not found"
+        )
     return response
 
 
-@api_router.delete("/books/{book_id}", tags=["books"])
-def delete_book(book_id: int, session: Session = Depends(get_session)):
-    response = controller.delete_book(book_id, session)
-    if type(response) == Err and response.error == NO_OBJECT_ERROR:
-        raise HTTPException(status_code=HTTPStatus.NOT_FOUND.value)
-    elif type(response) == Err:
-        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR.value)
+@api_router.put("/books/{book_id}", tags=["books"], response_model=models.Book)
+async def update_book(book_id: int, book: models.BookIn):
+    response = await controller.update_book(SQLiteRepository, book_id, book)
+    if isinstance(response, NotFoundError):
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND.value, detail="Book with such ID not found"
+        )
     return response
```

Comment:
It's good practice to have roots in some kind of router and files like xxx_router.py but it's not a big problem here

Reply:
Do you mean having a separate folder in the app called routers and inside a file e.g. books_router.py ? Not to have this code in main.py?

Reply:
exactly